test/app.spec.js

```diff
@@ -0,0 +1,72 @@
+import React from 'react';
+import { mount } from 'enzyme';
+import expect from 'expect';
+import App from '../src/containers/app';
+import configureMockStore from 'redux-mock-store';
+import { Provider } from 'react-redux';
+import Immutable from 'immutable';
+import fetchMock from 'fetch-mock';
+
+before(() => {
+  // If missing, fetch global will be undefined. The json-content is unused.
+  fetchMock.mock('^https://api-dev.floq.no', { unused: 'data' });
+});
+
+after(() => {
+  fetchMock.restore();
+});
+
+// setup is a function, so that each call gets it's own objects (mutable state)
+const setup = () => {
+  const params = {
+    id: 1
+  };
+
+  const projects = {
+    loading: false,
+    data: new Immutable.Map([
+      {
+        id: 1,
+        name: 'test_project_one',
+        customer: 'TEST1000'
+      }
+    ].map(e => [e.id, e]))
+  };
+
+  const customers = {
+    loading: false,
+    data: new Immutable.Map([
+      {
+        id: 'TEST1000',
+        name: 'test_customer_one'
+      },
+      {
+        id: 'TEST1001',
+        name: 'test_customer_two'
+      }
+    ].map(e => [e.id, e]))
+  };
+
+  const state = {
+    projects,
+    customers
+  };
+
+  const wrapper = mount(
+    <Provider store={configureMockStore([])(state)}>
+      <App
+        params={params}
+      />
+    </Provider>);
+
+  return {
+    wrapper
+  };
+};
+
+describe('<App />', () => {
+  it('contains detail-view class without children', () => {
+    const { wrapper } = setup();
+    expect(wrapper.find('.detail-view').props().children).toEqual(undefined);
```

Comment:
Correct; it gives little to no value. I just wanted to keep the example i
fought to get working :) Got any ideas to what is meaningful to test here?
There is little logic in this file.

2016-07-04 17:51 GMT+02:00 Erlend Hamberg <notifications@github.com>:

> In test/app.spec.js
> <https://github.com/blankoslo/floq-projects/pull/9#discussion_r69474461>:
>
> > +  const wrapper = mount(
> > +    <Provider store={configureMockStore([])(state)}>
> > +      <App
> > +        params={params}
> > +      />
> > +    </Provider>);
> > +
> > +  return {
> > +    wrapper
> > +  };
> > +};
> > +
> > +describe('<App />', () => {
> > +  it('contains detail-view class without children', () => {
> > +    const { wrapper } = setup();
> > +    expect(wrapper.find('.detail-view').props().children).toEqual(undefined);
>
> Maybe I'm misunderstanding something, but isn't this test completely
> specious? children is injected when using react-router, so of course it's
> undefined here. :)
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/blankoslo/floq-projects/pull/9/files/443304b4cedd6f1b9cbefd1e8af7e5e60896d796..98e1adb06ebd2ce149d2a4efc2548159608d0880#r69474461>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABCBjJCitLRYywqDuhj_KEL9ZOHsU3fBks5qSSvxgaJpZM4JEaBz>
> .
>